code/modules/mob/living/carbon/xenomorph/xenomorph.dm

```diff
@@ -411,3 +410,40 @@
 	SIGNAL_HANDLER
 	var/obj/alien/weeds/found_weed = locate(/obj/alien/weeds) in loc
 	loc_weeds_type = found_weed?.type
+
+/// Burrow code for xenomorphs
+/mob/living/carbon/xenomorph/proc/xeno_burrow()
+	SIGNAL_HANDLER
+	if(!burrowed)
+		to_chat(src, span_xenowarning("We start burrowing into the ground"))
```

Comment:
there should be a caste check before, to prevent admins from calling burrow to random xenos